kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/MagicCommand.java

```diff
@@ -140,11 +143,30 @@ public MagicCommandFunctionality addImport() {
       if (parts.length != 2) {
         return sendErrorMessage(message, "Wrong import format.", executionCount);
       }
-      this.kernel.addImport(new ImportPath(parts[1]));
-      return getMagicCommandItem(code, message, executionCount);
+
+      if (isValidImport(parts[1], executionCount)) {
+        this.kernel.addImport(new ImportPath(parts[1]));
+        return getMagicCommandItem(code, message, executionCount);
+      }
+
+      return sendErrorMessage(message, "Provided import doesn't exists. \n"
+          + "Try firstly add jar which contains class to classpath.", executionCount);
     };
   }
 
+  private boolean isValidImport(String part, int executionCount) {
+    try {
+      CompletableFuture<Boolean> validImportFuture = new CompletableFuture<>();
+      kernel.executeCode("Class.forName(\"" + part + "\")", new Message(), executionCount, seo -> {
```

Comment:
There should be a more more direct way of doing this that doesn't use executeCode like this.
For example, this will fail with the Clojure kernel.